### ISSUE TITLE
lyrics.loader.load_csv - tracking disable & noupdate

### ISSUE THREAD

**Issue by yvaucher** [camptocamp/anthem#42]
load_csv being mostly used to load tons of data, wouldn't it make sens to change the default values?

Usage has shown that we almost use tracking disabled every time.
Plus, data created from CSV is mostly not config and shouldn't be modified update, thus it should also
set the xml ids as `noupdate`. 

We can change the current function default behavior or have an alias that does this small change.

For me there is 2 use case:

- Most common: create new record data csv, data as the user create them, we don't want any module changing them (fake module name does the trick but it's hacky)
- Less common: Modify existing data, can be used for data that already exists in database, you don't care about noupdate because it is already in that state.

**Comment by blaggacao** [comment]
Check also GitHub.com/xoe-labs/dodoo-loader for additional ideas

**Comment by yvaucher** [comment]
@blaggacao could you be more specific. I don't see the link with the current proposal to change a feature of anthem lyrics by adding some default values in context.

**Comment by blaggacao** [comment]
@yvaucher sorry, if I confused you. There is no specific link. A loose association, for what it might be worth. ;) Best, David

**Comment by grindtildeath** [comment]
@yvaucher I just did a test on runbot 12.0, and must agree with @simahawk about "tracking_disable but not on noupdate".
Actually creating a record by CSV upload on the UI won't set this new record as noupdate. However the tracking is disabled as there are is no message posted. (http://399803-12-0-de76e5.runbot19.odoo.com/web?debug#id=2370&action=169&model=res.partner&view_type=form&menu_id=117)
Creating another record manually won't create any ir.model.data, but tracking is obviously activated in such case. (http://399803-12-0-de76e5.runbot19.odoo.com/web?debug#id=2371&action=169&model=res.partner&view_type=form&menu_id=117)

So for `load_csv` to be closer to odoo UI behaviour, we must set `tracking_disable=True`

**Comment by yvaucher** [comment]
> Actually creating a record by CSV upload on the UI won't set this new record as noupdate.

@grindtildeath this can lead to disaster when the user use existing module name in the xml ids of new records. For peace of mind I think it is preferable to set it noupdate by default.

**Comment by yvaucher** [comment]
Issue is that nothing forbids you to have a `__setup__` or `__export__` module.

![__setup__ module](https://user-images.githubusercontent.com/4158438/48706730-7101d980-ebfd-11e8-9d87-b5138be957a1.png)

Which could be disastrous if it was installed, or updated. Let's not forget that module's technical name is hidden and you can put whatever you want in the module name so you wouldn't notice it. Our current process should prevent us from falling in that trap, but I'm trying to picture the whole thing, and IMO all imported records should be noupdate by default.


But let's take aside the risk with special addons names which could be a malevolent act.  Let's answer your question:

> Do you have an example?

In case of `account.account` and `account.tax` creation, xmlid `namespace` is the name of the module. It works because the generation of those object set xmlids to noupdate.

Now, you will probably be exporting your created list of account.account and edit it in a csv file to reimport an edited version. There are some chance that, when editing it, you will simply want to copy / paste lines in the file, creating a new record. Or you would like to import it on a fresh install. Thus, you will have records with an xmlid on module `l10n_xx`. Those account and taxes will be deleted on next `l10n_xx` module update. And this is hardly detectable in a csv file.

IMO, it could be acceptable to have those new xmlid, for new accounts not originally created by the CoA install using the module namespace.
May I remind that we also had a case, of a change in the chart of account and to follow that change we made the import by hand forgetting about the noupdate. Other cases can be human errors when preparing the files. But, we could be error prone and limit the number of required checks on the CSV files.


> Setting noupdate would prevent further updates of the data by other CSV imports.

This is a wrong assumption, `noupdate` only affects the loading from a module update point of view. Loading a csv file through the interface will still modify your records even if those are tagged `noupdate`. 
You can try it with the product.categories, those base objects have a noupdate xmlid but you can still change them with a CSV import from the UI.


To sum up, having noupdate by default, avoids loss of data in some cases and has no other impact that I know of. Only actual module data shouldn't have noupdate, and thus should be dealt in modules.
Thus, with this advantage and no known issue, it would only ensure me some peace of mind.

**Comment by grindtildeath** [comment]
@yvaucher I opened #43 with a change to implement tracking_disable=True at the moment.
However, this context value is defined in `mail` module, and would not bring anything in case only `base` is installed. 
I'm wondering if it makes sense in such a case to 'tie' some marabunta code with a module that might not be installed all the time. :thinking:

**Comment by grindtildeath** [comment]
@yvaucher Was just asking, IMO let's not overcomplicate things and keep it simple.

**Comment by simahawk** [comment]
yeaah that's not even a dependency... an extra ctx key won't hurt.

> May I remind that we also had a case, of a change in the chart of account and to follow that change we made the import by hand forgetting about the noupdate. Other cases can be human errors when preparing the files. But, we could be error prone and limit the number of required checks on the CSV files.

That's why we should use DJ tool to unify exports and generation of such songs :wink: 
You can configure noupdate where you need it as well many other configs.
Also, you can enforce XIDs generation if you want to override existing ones.

**Comment by yvaucher** [comment]
@simahawk DJ tool is not related to anthem, as this tool is public I was thinking of something more user friendly with default values. I agree it could be disturbing to have different default behaviour but IMO it wouldn't be noticed and would be less destructive.
To enforce XIDs you need the record to exist to use add_xmlid.
